### ISSUE TITLE
[FEATURE] Testcontainers Cloud evaluation

### ISSUE THREAD

**Issue by eddumelendez** [langchain4j/langchain4j#544]
**Is your feature request related to a problem? Please describe.**
Use [Testcontainers Cloud](https://testcontainers.com/cloud/)

**Describe the solution you'd like**
[Testcontainers Cloud](https://testcontainers.com/cloud/) offers a [GHA](https://github.com/AtomicJar/testcontainers-cloud-setup-action) and can provide more worker to improve running test in parallel and the feedback loop.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by langchain4j** [comment]
I tried using it but it failed for ollama and localai...

**Comment by eddumelendez** [comment]
any specific issue you can share about? IIRC, I used Testcontainers Cloud to run project's test with Ollama and LocalAI.

**Comment by langchain4j** [comment]
Can't find the log... But here is where I disabled it because it was not working: https://github.com/langchain4j/langchain4j/pull/507/commits/56da23709502202975300d3db926d6a42c310bf6

I see multiple runs in the dashboard, but can't find how to see logs or any details:
![image](https://github.com/langchain4j/langchain4j/assets/132277850/29c2f50e-944c-47b7-a702-35d48458915a)

Account registered for langchain4j@gmail.com (in case you have access)

**Comment by langchain4j** [comment]
I will try to run it again in a few days and provide more details

**Comment by realjenius** [comment]
Hello! We did some testing on the Testcontainers Cloud side with the langchain4j test suite, and found at least one issue preventing both the `local-ai` and `ollama` modules from completing successfully consistently.

We've made some adjustments to hopefully help, and with the most recent changes I was able to run all the ITs from both of those modules against TCC -`local-ai` completed fine; there were some failures within the `ollama` integration tests, but they don't appear to be environment-specific.

I was also able to run the full suite as currently configured (with `-DskipITs`) against TCC without issue.

I would love to hear if your experience is any better with these changes!
